lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -1713,6 +1713,98 @@ static LogicalResult printOperation(CppEmitter &emitter,
   return success();
 }
 
+// Generate the comparison intrinsics(eq, ne, lt, le, gt, ge) for AIE-ML
+static LogicalResult printOperation(CppEmitter &emitter, aievec::CmpOp cmpOp) {
+  if (!AIEML) {
+    return failure();
+  }
+
+  // The lhs and rhs should have already been emitted
+  Value lhs = cmpOp.getLhs();
+  Value rhs = cmpOp.getRhs();
+
+  if (!emitter.hasValueInScope(lhs) || !emitter.hasValueInScope(rhs))
+    return failure();
+
+  // Generate the initialization for the vector
+  if (failed(emitter.emitAssignPrefix(*cmpOp)))
+    return failure();
+
+  raw_indented_ostream &os = emitter.ostream();
+
+  StringRef pred = cmpOp.getPred();
+  if (pred == "eq") {
+    os << "eq";
+  } else if (pred == "ne") {
+    os << "ne";
+  } else if (pred == "slt" || pred == "ult") {
+    os << "lt";
+  } else if (pred == "sle" || pred == "ule") {
+    os << "le";
+  } else if (pred == "sgt" || pred == "ugt") {
+    os << "gt";
+  } else if (pred == "sge" || pred == "uge") {
+    os << "ge";
+  } else {
+    return failure();
+  }
+
+  os << "(";
+  VectorType vType = lhs.getType().cast<VectorType>();
+  Type eltType = vType.getElementType();
+
+  if (eltType.isa<IntegerType>() &&
```

Comment:
Slight variants of  piece of code 1758-1772 appears already multiple times in this file and so it should be refactored.